rust/src/index/vector/pq.rs

```diff
@@ -26,14 +26,15 @@ use arrow_ord::sort::sort_to_indices;
 use arrow_schema::{DataType, Field as ArrowField, Schema as ArrowSchema};
 use arrow_select::take::take;
 
+use crate::index::pb;
 use crate::index::vector::kmeans::train_kmeans;
 use crate::io::object_reader::{read_fixed_stride_array, ObjectReader};
 use crate::Result;
 use crate::{arrow::*, utils::distance::l2_distance};
 
 /// Product Quantization Index.
 ///
-pub struct PQIndex {
+pub struct PQIndex<'a> {
```

Comment:
i still don't understand lifetimes. why is this needed here?

Reply:
PQIndex now does not own the `ProductQuantizer`, but just has a reference to the `ProductQuantizer`.  So PQIndex needs to make sure `ProductQuantizer` is alive.